### DISCUSSION TITLE
How to generate multiple `encodeurl` output files of all files from a directory?

### DISCUSSION THREAD

**Discussion by The-Lum** [plantuml/plantuml#1062] [Q&A]
Hello all,

_[To adding diagram links on the PlantUML Gallery]_
- How to generate multiple files with the `encodeurl` output of all files from a directory?

Hint:
```sh
java -jar plantuml.jar -o outputdir -encodeurl *.puml 
```
Currently that generates only all the `encodeurl` output on `stdout`, not on file!

Another alternative will be to does not use `plantuml.jar` and use another script language to generate `encodeurl` files?
_[Hints: [PlantUML text-encoding](http://plantuml.com/text-encoding), ...]_

**Comment by OnceUponALoop** [comment]
I was looking at doing something similar recently, I just extended my bash script to cover your use-case but you can also look at the python code, it might be easier to extend.

Hope this help :+1: 

![encode_pumls.sh](https://gist.github.com/OnceUponALoop/1dedfba8f75cf7beeea101121f8346bd)
```
Usage:
./encode_puml.sh -p <PLANTUML_JAR> -s <SEARCH_PATH> -o <OUTPUT_DIR> -f <FILE_PATTERN>

Options:
  -p <PLANTUML_JAR>  Path to the PlantUML jar file
  -s <SEARCH_PATH>   Path to search for files
  -o <OUTPUT_DIR>    Directory to save the encoded files
  -f <FILE_PATTERN>  File pattern to search for (e.g., "*.puml")
```

![plantuml_codec.py](https://gist.github.com/OnceUponALoop/a346aa9bb40ed1183d428ee753334c2d)
```
usage: plantuml_codec.py [-h] (-e | -d | -s) text

Encode or decode PlantUML text, or extract and decode from SVG.

positional arguments:
  text          The PlantUML text to encode, the encoded text to decode, or the SVG file path

options:
  -h, --help    show this help message and exit
  -e, --encode  Encode the provided PlantUML text
  -d, --decode  Decode the provided encoded PlantUML text
  -s, --svg     Extract and decode PlantUML text from the provided SVG file
```
